### PR TITLE
GH-39413: [C++][Parquet] Vectorize decode plain on FLBA

### LINKED ISSUE #39413: [C++][Parquet] DecodePlain<FixedLenByteArray> is not vectorized
### Describe the enhancement requested

In the link below i've written how vectorized version could be implemented
https://godbolt.org/z/xWeb93xjW

### Component(s)

C++, Parquet

### PR DESCRIPTION
**Pull request by Hattonuri**
### What changes are included in this PR?
FLBA Decode Plain is not vectorized. So this parsing can be implemented faster https://godbolt.org/z/xWeb93xjW

### Are these changes tested?
Yes, on unittest

### Are there any user-facing changes?

* Closes: #39413

### CODE REVIEW

**Comment by Hattonuri** [comment]
Actually i saw in binary that on real code it still is not vectorized.....

**Review by mapleFU — APPROVED** [review]
Aha I'm ok with this but I don't know why previous code failed to compile to SIMD, just do loop-unroll...

**Comment by mapleFU** [comment]
```
<source>:18:3: remark: loop not vectorized: value that could not be identified as reduction is used outside the loop [-Rpass-analysis=loop-vectorize]
   18 |   for (int i = 0; i < num_values; ++i) {
      |   ^
<source>:18:3: remark: loop not vectorized [-Rpass-missed=loop-vectorize]
```


With `-Rpass-analysis=loop-vectorize -Rpass=loop-vectorize -Rpass-missed=loop-vectorize`, the compiler reports why loop doesn't get vectorized

File: cpp/src/parquet/encoding.cc [hunk 1]
```diff
@@ -1080,9 +1080,7 @@ inline int DecodePlain<FixedLenByteArray>(const uint8_t* data, int64_t data_size
     ParquetException::EofException();
   }
   for (int i = 0; i < num_values; ++i) {
-    out[i].ptr = data;
-    data += type_length;
-    data_size -= type_length;
+    out[i].ptr = data + i * type_length;
```
Comment (mapleFU) [inline cpp/src/parquet/encoding.cc:1083]: oh I missed a thing, maybe I should cast the `i * type_length` to int64_t. I've do it in https://github.com/apache/arrow/pull/39420 . So sorry for that

**Pull request merged**